product_tier_validation/models/product_template.py

```diff
@@ -28,9 +28,3 @@ def write(self, vals):
         if "state" in vals:
             vals["active"] = self._is_tier_validated_active(vals["state"])
         return super().write(vals)
-
-    @api.model
-    def _get_default_product_state_id(self):
```

Comment:
this probably should be kept as is on v14

Reply:
As per my comment, we should keep the direct ref because we cannot create another default state.
Otherwise, you have to add migration and installation script to swap the default state on "draft" and trash this override here.